### DISCUSSION TITLE
Opt-Out from Packagist #StandWithUkraine Message

### DISCUSSION THREAD

**Discussion by ktomk** [composer/composer#10600] [Q&A / Support]
Regardless whether I personally find the changes introduced in #10582 inclusive or not, the current display running composer poses the question to me how to opt out from such messages technically.

Is this even possible?

/edit: As it appears more and more noisy to me, sharing an alias with redirection/process substitution and `unbuffer` to keep default colors behavior which seems to do the job on my box:

~~~
$ alias composer="> >(grep -av '#StandWith') unbuffer composer "    # eats exit code
$ alias composer="> >(grep -av '#StandWith') composer --ansi "      # preserves exit code but enforces colors, always
~~~

**Comment by mbomb007** [comment]
Here's a similar question and solution: https://stackoverflow.com/q/63731920

**Reply by ktomk** [reply]
@mbomb007  Thanks for the heads up, it indeed is of the same principle wrapping the call and filtering the output [modified]:

> ~~~
> $ alias composer="2> >(grep -av 'composer fund\|looking for funding') composer --ansi "
> ~~~

and with similar downsides. it's patching the same way, which is why I continue to think it would be useful if composer itself would offer more control on the messaging. Or to really wrap the composer binary within an expect/script pseudo pty layer (`unbuffer` is from expect if I got it right and `script` allows to perpetuate the exit status [also on Stackoverflow (SO)])

**Comment by Andreas-Schoenefeldt** [comment]
Here are also a couple of good solutions: https://github.com/composer/composer/discussions/10808

**Comment by mbomb007** [comment]
> Regardless whether I personally find the changes introduced in #10582 inclusive or not, the current display running composer poses the question to me how to opt out from such messages technically.
> 
> Is this even possible?
> 
> /edit: As it appears more and more noisy to me, sharing an alias with redirection/process substitution and `unbuffer` to keep default colors behavior which seems to do the job on my box:
> 
> ```
> $ alias composer="> >(grep -av '#StandWith') unbuffer composer "    # eats exit code
> $ alias composer="> >(grep -av '#StandWith') composer --ansi "      # preserves exit code but enforces colors, always
> ```

#10582 returns a 404 error. [View on archive.org](https://web.archive.org/web/20220325183037/https://github.com/composer/composer/issues/10582)

[Direct link to Packagist commit](https://github.com/composer/packagist/commit/86244a3695fcaaac9c5ba4257a4314eae1c6d981)

**Comment by politsin** [comment]
```sh
git clone https://github.com/composer/composer.git --branch 2.4.4  ~/composer-build && \
    composer install  -o -d ~/composer-build && \
    wget https://raw.githubusercontent.com/politsin/snipets/master/patch/composer.patch -q -O ~/composer-build/composer.patch  && \
    cd ~/composer-build && patch -p1 < composer.patch && \
    php -d phar.readonly=0 bin/compile && \
    rm /usr/local/bin/composer && \
    php composer.phar install && \
    php composer.phar update && \
    mv ~/composer-build/composer.phar /usr/local/bin/composer && \
    rm -rf ~/composer-build  && \
    chmod +x /usr/local/bin/composer
```
